google-cloud-spanner/src/main/java/com/google/cloud/spanner/ForwardingStructReader.java

```diff
@@ -20,14 +20,20 @@
 import com.google.cloud.Date;
 import com.google.cloud.Timestamp;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import java.util.List;
 
 /** Forwarding implements of StructReader */
 public class ForwardingStructReader implements StructReader {
 
-  private StructReader delegate;
+  private Supplier<? extends StructReader> delegate;
```

Comment:
This is to ensure that getting a session from the pool is always non-blocking. The client library hides all session management from the client application, but many of these classes are in one way or another a wrapper around a session. So in this specific case, instead of having to wait for the delegate `StructReader` to be available (i.e. for a `Session` to be returned from the pool), it is handed a supplier.